adview-manager/src/lib.rs

```diff
@@ -474,15 +468,17 @@ impl Manager {
                 if block_on(self.is_campaign_sticky(campaign.channel.id)) {
                     return vec![];
                 }
-                let mut campaign_input = targeting_input.clone();
+
                 let campaign_id = campaign.channel.id;
-                campaign_input.channel = Some(campaign.channel.clone());
+
+                let campaign_input = targeting_input.clone().with_market_channel(campaign.channel.clone());
 
                 campaign
                     .units_with_price
                     .iter()
                     .filter(|unit_with_price| {
-                        campaign_input.ad_unit = Some(unit_with_price.unit.clone());
+                        let mut unit_input = campaign_input.clone();
```

Comment:
is clone required here?

Reply:
It's not required indeed, good catch, thanks!

Reply:
:)